src/material/form-field/directives/floating-label.ts

```diff
@@ -43,6 +95,29 @@ export class MatFormFieldFloatingLabel {
   get element(): HTMLElement {
     return this._elementRef.nativeElement;
   }
+
+  /** Handles resize events from the ResizeObserver. */
+  private _handleResize() {
+    // In the case where the label grows in size, the following sequence of events occurs:
+    // 1. The label grows by 1px triggering the ResizeObserver
+    // 2. The notch is expanded to accommodate the entire label
+    // 3. The label expands to its full width, triggering the ResizeObserver again
+    //
+    // This is expected, but If we allow this to all happen within the same macro task it causes an
+    // error: `ResizeObserver loop limit exceeded`. Therefore we push the notch resize out until
+    // the next macro task.
+    setTimeout(() => this.resized.emit());
```

Comment:
I think that this going to cause all form fields to trigger a global change detection on init since the resize observer fires immediately. This might not have shown up in the perf test since the dev app is pretty basic.

Reply:
I refactored the resized output away and added a zone escape here, so I think we should be good now.